### PR TITLE
Fix inflection in TestTask to properly normalize first for consistent…

### PR DESCRIPTION
**Pull request by dereuromark**
… internal handling.

Resolves https://github.com/cakephp/bake/issues/437 and makes code way simpler and less error prone.

I recommend 1.8.0 release.

### CODE REVIEW

File: tests/TestCase/Shell/Task/TestTaskTest.php [hunk 1]
```diff
@@ -334,23 +334,23 @@ public static function realClassProvider()
...
+            ['Component', 'AuthComponent', 'App\Controller\Component\AuthComponent'],
             ['Shell', 'Example', 'App\Shell\ExampleShell'],
-            ['shell', 'Example', 'App\Shell\ExampleShell'],
+            ['Shell', 'Example', 'App\Shell\ExampleShell'],
+            ['Task', 'Example', 'App\Shell\Task\ExampleTask'],
             ['Task', 'Example', 'App\Shell\Task\ExampleTask'],
-            ['task', 'Example', 'App\Shell\Task\ExampleTask'],
             ['Cell', 'Example', 'App\View\Cell\ExampleCell'],
-            ['cell', 'Example', 'App\View\Cell\ExampleCell'],
+            ['Cell', 'Example', 'App\View\Cell\ExampleCell'],
```
Comment (markstory) [inline tests/TestCase/Shell/Task/TestTaskTest.php]: Looks like we have redundant tests now?

Comment (dereuromark) [reply tests/TestCase/Shell/Task/TestTaskTest.php]: Fixed up and added some more tests.

**Comment by ADmad** [comment]
This change was published in bugfix release 1.7.3 causing failures in community plugins which relied on this particular casing. Example:

https://github.com/HavokInspiration/wrench/blob/master/src/Shell/Task/MaintenanceModeTask.php#L59

**Comment by markstory** [comment]
We can work towards not having public properties in the future.

**Pull request merged**